sycl/include/sycl/backend.hpp

```diff
@@ -269,8 +287,23 @@ typename std::enable_if<
 make_queue(const typename backend_traits<Backend>::template input_type<queue>
                &BackendObject,
            const context &TargetContext, const async_handler Handler = {}) {
-  return detail::make_queue(detail::pi::cast<pi_native_handle>(BackendObject),
-                            TargetContext, nullptr, false, Handler, Backend);
+  if constexpr (Backend == backend::ext_oneapi_level_zero) {
+    bool IsImmCmdList = std::holds_alternative<ze_command_list_handle_t>(
+        BackendObject.NativeHandle);
+    pi_native_handle Handle =
+        IsImmCmdList ? reinterpret_cast<pi_native_handle>(
+                           *(std::get_if<ze_command_list_handle_t>(
+                               &BackendObject.NativeHandle)))
+                     : reinterpret_cast<pi_native_handle>(
+                           *(std::get_if<ze_command_queue_handle_t>(
+                               &BackendObject.NativeHandle)));
+    return sycl::detail::make_queue2(Handle, IsImmCmdList, TargetContext,
+                                     nullptr, false, BackendObject.Properties,
+                                     Handler, Backend);
+  } else {
```

Comment:
I think this code needs to be removed since you have proper specialization for level_zero backend here: https://github.com/intel/llvm/pull/7196/files#diff-bad45c4e6d46c89d807c929b6adbffdb70993c102647c58b41fb0534e2195727R121

Reply:
Yes, it had been removed, but I forgot to push the change to that file.
I think some other files also need to be updated. I am working on updating all files properly now.

Reply:
Code has been removed.